packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts

```diff
@@ -49,21 +49,22 @@ export default function deepCyclicCopyReplaceable<T>(
 
 function deepCyclicCopyObject<T>(object: T, cycles: WeakMap<any, any>): T {
   const newObject = Object.create(Object.getPrototypeOf(object));
-  const descriptors = Object.getOwnPropertyDescriptors(object);
+  const descriptors: {
+    [x: string]: PropertyDescriptor;
+  } = Object.getOwnPropertyDescriptors(object);
 
   cycles.set(object, newObject);
 
   Object.keys(descriptors).forEach(key => {
-    const descriptor = descriptors[key];
-    if (typeof descriptor.value !== 'undefined') {
-      descriptor.value = deepCyclicCopyReplaceable(descriptor.value, cycles);
-    }
-
-    if (!('set' in descriptor)) {
-      descriptor.writable = true;
-    }
-
-    descriptor.configurable = true;
+    descriptors[key] = {
+      configurable: true,
+      enumerable: true,
+      value: deepCyclicCopyReplaceable(
+        (object as Record<string, unknown>)[key],
```

Comment:
without casting
![image](https://user-images.githubusercontent.com/1404810/78281277-b84ee880-751a-11ea-9de7-d61d4c32b6cf.png)

not sure if it's solvable?